source/text.cpp

```diff
@@ -530,11 +542,35 @@ spv_result_t spvTextEncodeOpcode(const libspirv::AssemblyGrammar& grammar,
   // NOTE: The table contains Opcode names without the "Op" prefix.
   const char* pInstName = opcodeName.data() + 2;
 
-  spv_opcode_desc opcodeEntry;
-  error = grammar.lookupOpcode(pInstName, &opcodeEntry);
-  if (error) {
-    return context->diagnostic(error) << "Invalid Opcode name '" << opcodeName
-                                      << "'";
+  spv_opcode_desc opcodeEntry = nullptr;
+
+  // If it's an unknown opcode, then store its description the object
+  // owned by this unique pointer.  It's ok for the object to be deleted at the
+  // end of this scope since we won't need it any more.
+  std::unique_ptr<spv_opcode_desc_t> unknown_opcode_desc = nullptr;
+  // First see if it's parseable as a non-zero 16 bit number. OpNop has
+  // opcode zero, and if you want that then just say OpNop instead of Op0.
+  if (auto unknown_opcode = As16BitNumber(pInstName)) {
+    // Support this as an unknown instruction.  Don't make an opcode name
+    // for it.  Assume it:
+    //  - does not have a type
+    //  - does not generate a result Id
+    //  - has zero or more literal number operands.
+    unknown_opcode_desc.reset(
```

Comment:
I think if we gonna do this, it's better to check whether this number is the opcode value for some known instruction. 

This feature itself is already very scary. Instead of leaving the users totally unprotected, we should try our best to guard them against some silly mistakes.